ilastik/workflows/carving/opPreprocessing.py

```diff
@@ -431,19 +430,16 @@ def execute(self, slot, subindex, roi, result):
         # Cache result
         self._prepData = result
 
-        # Wonder why this is set?
-        # The preprocess is only called by the run button.
-        # By setting the output dirty, this event is propagated to the
-        # carving-Operator, who copies the result just calculated.
-        # This is to gain control over when the preprocess is executed.
-        # self.PreprocessedData.setDirty()
-
         result[0] = mst
         return result
 
     def AreSettingsInitial(self):
         """analyse settings for sigma and filter
         return True if they are equal to those of last preprocess"""
+        warnings.warn(
```

Comment:
Warnings are not displayed to end users, and we don't systematically collect logs. If we really want to be thorough when removing old, but suspicious code, we somehow need to display these messages to the end user somewhere in GUI, but in a way that doesn't disturb their normal activities.

Reply:
I'd argue that  a warning is fine here. We (as ilastik developers) should make sure it is not called in our code (and as far as I can tell, it isn't). The warning would be for _other_ developers that might rely on this function somehow. So warning is the way to go.

Reply:
Ah sorry, could have taken this back out. I'm throwing this entire function out anyway in the next PR because it's clear this is unused, and it's also clear where it came from and that it's no longer needed (the workflow used to have a "Reset" button back in 2013).